pkg/operation/operation.go

```diff
@@ -284,11 +284,28 @@ func (b *Builder) Build(ctx context.Context, clientMap clientmap.ClientMap) (*Op
 	}
 	operation.Shoot = shoot
 
-	shootedSeed, err := gardencorev1beta1helper.ReadShootedSeed(shoot.Info)
+	// Get the ManagedSeed object for this shoot, if it exists.
+	// Also read the managed seed API server settings from the managed-seed-api-server annotation.
+	operation.ManagedSeed, err = kutil.GetManagedSeed(ctx, gardenClient.GardenSeedManagement(), shoot.Info.Namespace, shoot.Info.Name)
 	if err != nil {
-		logger.Warnf("Cannot use shoot %s/%s as shooted seed: %+v", shoot.Info.Namespace, shoot.Info.Name, err)
-	} else {
-		operation.ShootedSeed = shootedSeed
+		logger.Warnf("Cannot use shoot %s/%s as managed seed: %+v", shoot.Info.Namespace, shoot.Info.Name, err)
```

Comment:
Should we return an error here?

Reply:
Are you sure? There was no such error before, so it was possible to reconcile a shoot even if the syntax of the `use-as-seed` annotation was wrong. I believe this was intentional, and so I preserved this behavior. Now 3 different things can go wrong:

* The `ManagedSeed` object exists but can't be read for some reason.
* The syntax of the new `managed-seed-api-server` annotation is wrong.
* The syntax of the old `use-as-seed` annotation is wrong.

Do we want to fail the shoot reconciliation in all these 3 cases, ignoring compatibility considerations? This means that some shoots that were successfully reconciled before will now stop reconciling. To be safe, I would think we would want to preserve the old behavior or perhaps only fail in the first case.

Reply:
Hmm, all true what you say.
Though, I would argue that it's always better to fail early, report the encountered errors and so on and rather let the user fix the misconfiguration instead of ignoring it. Error logs will most probably not be read by operators, unless they notice anything suspicious.
I would additionally argue, that this behavior is a technical debt of the old implementation and we should rather properly report errors now that we have proper resources and try to cleanup all of this mess. 
I think this `incompatibility` shouldn't hurt too much, as the `use-as-seed` was anyways only usable in the `garden` namespace so that enduser clusters should not be affected and operators should/will anyways look after their seed clusters.

Reply:
OK, I agree actually, just wanted to be cautious. If there are no objections raised by other reviewers I will do as you suggest.

Reply:
Done as suggested